### PR TITLE
AWS OIDC: DeployService service

### PR DESCRIPTION
**Pull request by marcoandredinis**
This PR creates a new method on the AWS OIDC gRPC service that deploys an ECS Service.

This is part of a refactor that moves the API calls behind the Auth Service.

Context: https://github.com/gravitational/teleport/issues/37245

### CODE REVIEW

File: api/proto/teleport/integration/v1/awsoidc_service.proto [hunk 1]
```diff
@@ -236,6 +239,53 @@ message DeployDatabaseServiceResponse {
...
+  // Eg, 14.3.4 (no "v" prefix)
+  // Required.
+  string teleport_version = 7;
+  // DeploymentJoinTokenName is the Teleport IAM Join Token to be used by the deployed
+  // service to join the cluster.
+  // Required.
+  string deployment_join_token_name = 8;
+  // DeployServiceConfig is the teleport.yaml configuration (base64 encoded) used by teleport.
+  // Required.
+  string deploy_service_config = 9;
```
Comment (justinas) [inline api/proto/teleport/integration/v1/awsoidc_service.proto:274]: Nit: `teleport_config_string` would be consistent with `DeployDatabaseServiceDeployment` and probably a more descriptive name for this.

File: lib/auth/integration/integrationv1/awsoidc.go [hunk 2]
```diff
@@ -417,6 +417,56 @@ func (s *AWSOIDCService) DeployDatabaseService(ctx context.Context, req *integra
 	}, nil
 }
 
+// DeployService deploys Services into Amazon ECS.
+func (s *AWSOIDCService) DeployService(ctx context.Context, req *integrationpb.DeployServiceRequest) (*integrationpb.DeployServiceResponse, error) {
```
Comment (justinas) [inline lib/auth/integration/integrationv1/awsoidc.go:421]: This probably needs a unit-test checking at least the auth part.

File: api/proto/teleport/integration/v1/awsoidc_service.proto (hunk 1 above)
Comment (marcoandredinis) [reply api/proto/teleport/integration/v1/awsoidc_service.proto]: Yeah, looks better.
Changed

File: lib/auth/integration/integrationv1/awsoidc.go (hunk 2 above)
Comment (marcoandredinis) [reply lib/auth/integration/integrationv1/awsoidc.go:421]: Added two tests for this to ensure RBAC is applied

**Review by justinas — APPROVED** [review]

File: e
Comment (kimlisa) [inline e:1]: did you mean to include this?

**Review by kimlisa — APPROVED** [review]

**Pull request merged**